### ISSUE TITLE
Calculating information content from different datasets

### ISSUE THREAD

**Issue by icooperstein** [Centogene/pyhpo#6]
Hello,

I have been using pyHPO to calculate the similarity scores between patients within a dataset using their clinical phenotype lists as HPOSets. ie - `HPOSet1.similarity(HPOSet2)`
However, I am worried my analyses may be skewed because pyHPO calculates the information contents used in the scoring algorithms based on  the "kind" parameter - OMIM, orpha, decipher, or gene. I am wondering if there is any way to create a "custom kind" of sorts so that my patients' similarity scores are calculated using information contents derived from my dataset of choice instead of these publicly available ones? 

Any feedback would be greatly appreciated. Thanks!

**Comment by anergictcell** [comment]
Hi, 

At the moment PyHPO does not support custom `information_content` values. But that would be a good feature to add, doesn't seem too difficult.

In the meantime, you could write your own `Similarity` method (see the example here: https://centogene.github.io/pyhpo/similarity.html).

Alternatively, you could take a hacky approach and overwrite one of the existing `information_content` in every term. So, you could do something like this:

```python
from pyhpo import Ontology
_ = Ontology()
for term in Ontology:
    term.information_content.decipher = my_ic_calculating_function(term)

...
HPOSet1.similarity(HPOSet2, kind='decipher')
```

**Comment by anergictcell** [comment]
Dear @icooperstein,

We have now released v3.1.2 for PyHPO, which allows adding custom information content to each HPO term. Thanks for the feature suggestion.
You can check here for a brief documentation:
https://centogene.github.io/pyhpo/hpoterm.html#set-custom

I hope PyHPO continues to be useful for your research.
